### ISSUE TITLE
Intermittent failures in simd_op_check

### ISSUE THREAD

**Issue by steven-johnson** [halide/Halide#3669]
e.g., https://buildbot.halide-lang.org/master/#/builders/39/builds/340

```
The vector and scalar versions of op_ld3_456 disagree. Maximum error: 4
../halide/Makefile:1656: recipe for target 'quiet_correctness_simd_op_check' failed

```

**Comment by steven-johnson** [comment]
FYI, I happened to get one of the intermittent failures locally (x86-64-osx), and captured the assembly output:

```
The vector and scalar versions of op_shufps_182 disagree. Maximum error: 1.64589e+14
Error assembly:
  .section  __TEXT,__text,regular,pure_instructions
  .section  __TEXT,__literal4,4byte_literals
  .p2align  2               ## -- Begin function error_op_shufps_182
LCPI0_0:
  .long 4286578688              ## float -Inf
  .section  __TEXT,__text,regular,pure_instructions
  .globl  _error_op_shufps_182
  .p2align  4, 0x90
_error_op_shufps_182:                   ## @error_op_shufps_182
  .cfi_startproc
## %bb.0:                               ## %entry
  pushq %r15
  .cfi_def_cfa_offset 16
  pushq %r14
  .cfi_def_cfa_offset 24
  pushq %r13
  .cfi_def_cfa_offset 32
  pushq %r12
  .cfi_def_cfa_offset 40
  pushq %rbx
  .cfi_def_cfa_offset 48
  subq  $16, %rsp
  .cfi_def_cfa_offset 64
  .cfi_offset %rbx, -48
  .cfi_offset %r12, -40
  .cfi_offset %r13, -32
  .cfi_offset %r14, -24
  .cfi_offset %r15, -16
  movq  96(%rsp), %rax
  movq  16(%rax), %r15
  movq  16(%rdi), %rbx
  xorl  %r12d, %r12d
  movl  $393220, %esi           ## imm = 0x60004
  xorl  %edi, %edi
  callq _halide_malloc
  movq  %rax, %r14
  .p2align  4, 0x90
LBB0_1:                                 ## %"for op_shufps_182$1.s0.y"
                                        ## =>This Loop Header: Depth=1
                                        ##     Child Loop BB0_2 Depth 2
  xorl  %ecx, %ecx
  .p2align  4, 0x90
LBB0_2:                                 ## %"for op_shufps_182$1.s0.x.x"
                                        ##   Parent Loop BB0_1 Depth=1
                                        ## =>  This Inner Loop Header: Depth=2
  vmovaps 512(%rbx,%rcx,2), %xmm0
  vshufps $216, 524(%rbx,%rcx,2), %xmm0, %xmm0 ## xmm0 = xmm0[0,2],mem[1,3]
  vmovaps %xmm0, (%rax,%rcx)
  addq  $16, %rcx
  cmpq  $3072, %rcx             ## imm = 0xC00
  jne LBB0_2
## %bb.3:                               ## %"end for op_shufps_182$1.s0.x.x"
                                        ##   in Loop: Header=BB0_1 Depth=1
  incq  %r12
  addq  $3072, %rax             ## imm = 0xC00
  cmpq  $128, %r12
  jne LBB0_1
## %bb.4:                               ## %"end for op_shufps_182$1.s0.y"
  xorl  %r13d, %r13d
  movl  $393220, %esi           ## imm = 0x60004
  xorl  %edi, %edi
  callq _halide_malloc
  movq  %rax, %r12
  .p2align  4, 0x90
LBB0_5:                                 ## %"for scalar_op_shufps_182.s0.y"
                                        ## =>This Loop Header: Depth=1
                                        ##     Child Loop BB0_6 Depth 2
  xorl  %ecx, %ecx
  .p2align  4, 0x90
LBB0_6:                                 ## %"for scalar_op_shufps_182.s0.x"
                                        ##   Parent Loop BB0_5 Depth=1
                                        ## =>  This Inner Loop Header: Depth=2
  movl  512(%rbx,%rcx,8), %edx
  movl  %edx, (%rax,%rcx,4)
  incq  %rcx
  cmpq  $768, %rcx              ## imm = 0x300
  jne LBB0_6
## %bb.7:                               ## %"end for scalar_op_shufps_182.s0.x"
                                        ##   in Loop: Header=BB0_5 Depth=1
  incq  %r13
  addq  $3072, %rax             ## imm = 0xC00
  cmpq  $128, %r13
  jne LBB0_5
## %bb.8:                               ## %"for maximum$182.s1.r$182$y.preheader"
  vmovss  LCPI0_0(%rip), %xmm4    ## xmm4 = mem[0],zero,zero,zero
  xorl  %eax, %eax
  vxorps  %xmm0, %xmm0, %xmm0
  movq  %r14, %rcx
  movq  %r12, %rdx
  .p2align  4, 0x90
LBB0_9:                                 ## %"for maximum$182.s1.r$182$y"
                                        ## =>This Loop Header: Depth=1
                                        ##     Child Loop BB0_10 Depth 2
  xorl  %esi, %esi
  .p2align  4, 0x90
LBB0_10:                                ## %"for maximum$182.s1.r$182$x"
                                        ##   Parent Loop BB0_9 Depth=1
                                        ## =>  This Inner Loop Header: Depth=2
  vmovss  (%rcx,%rsi,4), %xmm1    ## xmm1 = mem[0],zero,zero,zero
  vmovss  (%rdx,%rsi,4), %xmm2    ## xmm2 = mem[0],zero,zero,zero
  vsubss  %xmm2, %xmm1, %xmm3
  vsubss  %xmm1, %xmm2, %xmm1
  vcmpless  %xmm3, %xmm0, %xmm2
  vblendvps %xmm2, %xmm3, %xmm1, %xmm1
  vmaxss  %xmm4, %xmm1, %xmm4
  incq  %rsi
  cmpq  $768, %rsi              ## imm = 0x300
  jne LBB0_10
## %bb.11:                              ## %"end for maximum$182.s1.r$182$x"
                                        ##   in Loop: Header=BB0_9 Depth=1
  incq  %rax
  addq  $3072, %rdx             ## imm = 0xC00
  addq  $3072, %rcx             ## imm = 0xC00
  cmpq  $128, %rax
  jne LBB0_9
## %bb.12:                              ## %call_destructor.exit
  xorl  %edi, %edi
  movq  %r14, %rsi
  vmovss  %xmm4, 12(%rsp)         ## 4-byte Spill
  callq _halide_free
  xorl  %edi, %edi
  movq  %r12, %rsi
  callq _halide_free
  vmovss  12(%rsp), %xmm0         ## 4-byte Reload
                                        ## xmm0 = mem[0],zero,zero,zero
  vcvtss2sd %xmm0, %xmm0, %xmm0
  vmovsd  %xmm0, (%r15)
  xorl  %eax, %eax
  addq  $16, %rsp
  popq  %rbx
  popq  %r12
  popq  %r13
  popq  %r14
  popq  %r15
  retq
  .cfi_endproc
                                        ## -- End function
  .globl  _error_op_shufps_182_argv ## -- Begin function error_op_shufps_182_argv
  .p2align  4, 0x90
_error_op_shufps_182_argv:              ## @error_op_shufps_182_argv
  .cfi_startproc
## %bb.0:                               ## %entry
  subq  $40, %rsp
  .cfi_def_cfa_offset 48
  movq  (%rdi), %rax
  movq  80(%rdi), %rcx
  movq  %rcx, 32(%rsp)
  movq  %rax, %rdi
  callq _error_op_shufps_182
  xorl  %eax, %eax
  addq  $40, %rsp
  retq
  .cfi_endproc
                                        ## -- End function
  .globl  _error_op_shufps_182_metadata ## -- Begin function error_op_shufps_182_metadata
  .p2align  4, 0x90
_error_op_shufps_182_metadata:          ## @error_op_shufps_182_metadata
## %bb.0:                               ## %entry
  leaq  l_error_op_shufps_182_metadata_storage(%rip), %rax
  retq
                                        ## -- End function
  .section  __TEXT,__const
  .p2align  3               ## @0
l___unnamed_1:
  .space  16

  .p2align  5               ## @str
l_str:
  .asciz  "in_f32"

  .p2align  3               ## @1
l___unnamed_2:
  .space  16

  .p2align  5               ## @str.3
l_str.3:
  .asciz  "in_f64"

  .p2align  3               ## @2
l___unnamed_3:
  .space  16

  .p2align  5               ## @str.4
l_str.4:
  .asciz  "in_i8"

  .p2align  3               ## @3
l___unnamed_4:
  .space  16

  .p2align  5               ## @str.5
l_str.5:
  .asciz  "in_u8"

  .p2align  3               ## @4
l___unnamed_5:
  .space  16

  .p2align  5               ## @str.6
l_str.6:
  .asciz  "in_i16"

  .p2align  3               ## @5
l___unnamed_6:
  .space  16

  .p2align  5               ## @str.7
l_str.7:
  .asciz  "in_u16"

  .p2align  3               ## @6
l___unnamed_7:
  .space  16

  .p2align  5               ## @str.8
l_str.8:
  .asciz  "in_i32"

  .p2align  3               ## @7
l___unnamed_8:
  .space  16

  .p2align  5               ## @str.9
l_str.9:
  .asciz  "in_u32"

  .p2align  3               ## @8
l___unnamed_9:
  .space  16

  .p2align  5               ## @str.10
l_str.10:
  .asciz  "in_i64"

  .p2align  3               ## @9
l___unnamed_10:
  .space  16

  .p2align  5               ## @str.11
l_str.11:
  .asciz  "in_u64"

  .p2align  5               ## @str.12
l_str.12:
  .asciz  "error_op_shufps_182"

  .section  __DATA,__const
  .p2align  4               ## @10
l___unnamed_11:
  .quad l_str
  .long 1                       ## 0x1
  .long 1                       ## 0x1
  .byte 2                       ## 0x2
  .byte 32                      ## 0x20
  .short  1                       ## 0x1
  .space  4
  .quad 0
  .quad 0
  .quad 0
  .quad 0
  .quad l___unnamed_1
  .quad l_str.3
  .long 1                       ## 0x1
  .long 1                       ## 0x1
  .byte 2                       ## 0x2
  .byte 64                      ## 0x40
  .short  1                       ## 0x1
  .space  4
  .quad 0
  .quad 0
  .quad 0
  .quad 0
  .quad l___unnamed_2
  .quad l_str.4
  .long 1                       ## 0x1
  .long 1                       ## 0x1
  .byte 0                       ## 0x0
  .byte 8                       ## 0x8
  .short  1                       ## 0x1
  .space  4
  .quad 0
  .quad 0
  .quad 0
  .quad 0
  .quad l___unnamed_3
  .quad l_str.5
  .long 1                       ## 0x1
  .long 1                       ## 0x1
  .byte 1                       ## 0x1
  .byte 8                       ## 0x8
  .short  1                       ## 0x1
  .space  4
  .quad 0
  .quad 0
  .quad 0
  .quad 0
  .quad l___unnamed_4
  .quad l_str.6
  .long 1                       ## 0x1
  .long 1                       ## 0x1
  .byte 0                       ## 0x0
  .byte 16                      ## 0x10
  .short  1                       ## 0x1
  .space  4
  .quad 0
  .quad 0
  .quad 0
  .quad 0
  .quad l___unnamed_5
  .quad l_str.7
  .long 1                       ## 0x1
  .long 1                       ## 0x1
  .byte 1                       ## 0x1
  .byte 16                      ## 0x10
  .short  1                       ## 0x1
  .space  4
  .quad 0
  .quad 0
  .quad 0
  .quad 0
  .quad l___unnamed_6
  .quad l_str.8
  .long 1                       ## 0x1
  .long 1                       ## 0x1
  .byte 0                       ## 0x0
  .byte 32                      ## 0x20
  .short  1                       ## 0x1
  .space  4
  .quad 0
  .quad 0
  .quad 0
  .quad 0
  .quad l___unnamed_7
  .quad l_str.9
  .long 1                       ## 0x1
  .long 1                       ## 0x1
  .byte 1                       ## 0x1
  .byte 32                      ## 0x20
  .short  1                       ## 0x1
  .space  4
  .quad 0
  .quad 0
  .quad 0
  .quad 0
  .quad l___unnamed_8
  .quad l_str.10
  .long 1                       ## 0x1
  .long 1                       ## 0x1
  .byte 0                       ## 0x0
  .byte 64                      ## 0x40
  .short  1                       ## 0x1
  .space  4
  .quad 0
  .quad 0
  .quad 0
  .quad 0
  .quad l___unnamed_9
  .quad l_str.11
  .long 1                       ## 0x1
  .long 1                       ## 0x1
  .byte 1                       ## 0x1
  .byte 64                      ## 0x40
  .short  1                       ## 0x1
  .space  4
  .quad 0
  .quad 0
  .quad 0
  .quad 0
  .quad l___unnamed_10
  .quad l_str.12
  .long 2                       ## 0x2
  .long 0                       ## 0x0
  .byte 2                       ## 0x2
  .byte 64                      ## 0x40
  .short  1                       ## 0x1
  .space  4
  .quad 0
  .quad 0
  .quad 0
  .quad 0
  .quad 0

  .section  __TEXT,__const
  .p2align  5               ## @str.13
l_str.13:
  .asciz  "x86-64-osx-avx-avx2-disable_llvm_loop_unroll-disable_llvm_loop_vectorize-f16c-fma-no_asserts-no_bounds_query-no_runtime-sse41"

  .section  __DATA,__const
  .p2align  4               ## @error_op_shufps_182_metadata_storage
l_error_op_shufps_182_metadata_storage:
  .long 1                       ## 0x1
  .long 11                      ## 0xb
  .quad l___unnamed_11
  .quad l_str.13
  .quad l_str.12


.subsections_via_symbols
```

**Comment by steven-johnson** [comment]
FYI: I re-ran this several times on my system, force-dumping the same assembly regardless of errors. The 'correct' output is identical to the captured one above (aside from trivial label names in comments), so something else is afoot:

- input/output buffer contents unreliable or corrupted?
- the saved assembly doesn't accurately reflect what is actually being executed?
- (shouldn't be a threading issue as this test is being run single-threaded)

**Comment by steven-johnson** [comment]
Closing as this turns out to be a known thread-unsafeness issue that simd_op_check was prone to; it was band-aid fixed via #3914.
